src/WriteStream.h

```diff
@@ -30,9 +30,9 @@ using json11::Json;
 
 class WriteStream {
 public:
-    WriteStream(const IODirectives& ioDirectives, const SymbolTable& symbolTable,
-            const RecordTable& recordTable, bool summary = false)
-            : symbolTable(symbolTable), recordTable(recordTable), summary(summary) {
+    WriteStream(
+            const IODirectives& ioDirectives, const SymbolTable& symbolTable, const RecordTable& recordTable)
+            : symbolTable(symbolTable), recordTable(recordTable), summary(getSummary(ioDirectives)) {
```

Comment:
Instead of `getSummary(ioDirectives)` you could use `ioDirectives.getIOType() == "stdoutprintsize"`.